### PR TITLE
Add tests for stackTrace, scopes, variables

### PR DESCRIPTION
**Pull request by jtpio**
Add several tests for `stackTrace`, `scopes` and `variablesReference`.

They can also serve as a reference for the protocol messages.

### CODE REVIEW

**Comment by jtpio** [comment]
This change also adds an `execute` method to the `DebugSession`, used to send a `request_execute` message to the kernel and start debugging the code.

**Comment by jtpio** [comment]
This could be also rebased on https://github.com/jupyterlab/debugger/pull/15 if we get it in first.

File: src/session.ts [hunk 1]
```diff
@@ -90,6 +93,13 @@ export class DebugSession implements IDebugger.ISession {
     });
   }
 
+  /**
+   * Request code execution.
+   */
+  async execute(code: string): Promise<void> {
+    void this._sendExecuteMessage({ code });
```
Comment (afshin) [inline src/session.ts]: Should this be `await` instead of `void`?

Comment (jtpio) [reply src/session.ts]: Right, good catch!

File: src/session.ts [hunk 2]
```diff
@@ -147,8 +157,27 @@ export class DebugSession implements IDebugger.ISession {
...
+  private async _sendExecuteMessage(
+    msg: KernelMessage.IExecuteRequestMsg['content']
+  ): Promise<void> {
+    const kernel = this.client.kernel;
+    if (this._executeFuture) {
+      this._executeFuture.dispose();
+    }
+    this._executeFuture = kernel.requestExecute(msg);
+    this._executeFuture.onReply = (msg: KernelMessage.IExecuteReplyMsg) => {};
+  }
```
Comment (afshin) [inline src/session.ts]: I am confused about the promise this method is returning, in light of my last comment.

What is this promise guaranteeing when it resolves?

Comment (jtpio) [reply src/session.ts]: This should probably be handled in a better way indeed or simplified.

The idea is that an `execute_request` is sent to the kernel to start the debugging session. But the reply message will be received only when all threads have continued.

Also I would say the reply message (`execute_reply`) can be ignored for now.

Comment (jtpio) [reply src/session.ts]: Actually the `execute_request` should be triggered by a user (ctrl-enter a cell), so the `DebugSession` doesn't need this `execute` method.

In the tests we can trigger an `execute_request` by using the kernel directly:

`client.kernel.requestExecute({ code })`

**Comment by afshin** [comment]
Hm, the tests hang for me:

Screen Shot 2019-08-22 at 11 54 19 AM

**Comment by jtpio** [comment]
Looks like it cannot find xeus-python and start it.

**Comment by jtpio** [comment]
Maybe we should make the tests install the spec for `xpython` to the `data_dir` used for the tests instead: https://github.com/jupyterlab/jupyterlab/blob/f1bc02ced61881df94c49929837c49c022f5b115/jupyterlab/tests/test_jupyterlab.py#L126

**Comment by afshin** [comment]
That sounds like a good idea. Hm, I have the kernel when I run lab ...

**Comment by jtpio** [comment]
The last commit installs the kernel spec for xeus-python automatically when running the tests.

Opened https://github.com/jupyterlab/jupyterlab/pull/7089 to make it easier to do the same in other third-party extensions.

**Comment by jtpio** [comment]
@afshin @KsavinN what do you think about getting this one in?

Then we can take care of https://github.com/jupyterlab/debugger/issues/32 and start integrating the debug messages with the UI.

**Pull request merged**